spyder_kernels/comms/commbase.py

```diff
@@ -0,0 +1,407 @@
+        comm_id: int
+            the comm to send to. If None sends to all comms.
+        """
+
+        if not self.is_open(comm_id):
+            raise CommError("The comm is not connected.")
+        import cloudpickle
+        msg_dict = {
+            'spyder_msg_type': spyder_msg_type,
+            'content': content,
+            }
+        buffers = [cloudpickle.dumps(data, protocol=PICKLE_PROTOCOL)]
+        # Needed to prevent memory leaks. See spyder-ide/spyder#7158.
+        if len(buffers[0]) > MAX_SERIALIZED_LENGHT:
+            raise RuntimeError("Maximum serialized length exceeded.")
+        if comm_id is None:
+            # send to all the comms
+            id_list = list(self._comms.keys())
+        else:
+            id_list = [comm_id]
+        for comm_id in id_list:
+            self._comms[comm_id].send(msg_dict, buffers=buffers)
+
+    @property
+    def _comm_name(self):
+        """
+        Get the name used for the underlying comms.
+        """
+        return 'spyder_api'
+
+    def _register_message_handler(self, message_id, handler):
+        """
+        Register a message handler.
+
+        Parameters
+        ----------
+        message_id : str
+            The identifier for the message
+        handler : callback
+            A function to handle the message. This is called with 3 arguments:
+                - msg_dict: A dictionary with message information.
+                - buffer: The data transmitted in the buffer
+                - load_exception: Exception from buffer unpickling
+            Pass None to unregister the message_id
+        """
+        if handler is None:
+            self._message_handlers.pop(message_id, None)
+            return
+
+        self._message_handlers[message_id] = handler
+
+    def _register_comm(self, comm):
+        """
+        Open a new comm to the kernel.
+        """
+        comm.on_msg(self._comm_message)
+        comm.on_close(self._comm_close)
+        self._comms[comm.comm_id] = comm
+
+    def _comm_close(self, msg):
+        """Close comm."""
+        comm_id = msg['content']['comm_id']
+        del self._comms[comm_id]
+
+    def _comm_message(self, msg):
+        """
+        Handle internal spyder messages.
+        """
+        self.calling_comm_id = msg['content']['comm_id']
+        # Load the buffer. Only one is supported.
+        try:
+            if PY2:
+                buffer = cloudpickle.loads(msg['buffers'][0])
+            else:
+                buffer = cloudpickle.loads(bytes(msg['buffers'][0]))
+            load_exception = None
+        except Exception as e:
+            load_exception = e
+            buffer = None
+
+        # Get message dict
+        msg_dict = msg['content']['data']
+
+        spyder_msg_type = msg_dict['spyder_msg_type']
+
+        if spyder_msg_type in self._message_handlers:
+            self._message_handlers[spyder_msg_type](
+                msg_dict, buffer, load_exception)
+            return
```

Comment:
Let's remove this `return` here and instead add an `else` in this line, like this:

```python
else:
    logger.debug("No such spyder message type: %s" % spyder_msg_type)
```

Reply:
Here, I log the error instead of raising it because this is just a message, not a function call.